python/pyspark/sql/session.py

```diff
@@ -1308,7 +1308,7 @@ def prepare(obj: Any) -> Any:
         df._schema = struct
         return df
 
-    def sql(self, sqlQuery: str, args: Dict[str, str] = {}, **kwargs: Any) -> DataFrame:
+    def sql(self, sqlQuery: str, args: Optional[Dict[str, str]] = None, **kwargs: Any) -> DataFrame:
```

Comment:
Also fix the default parameter to `None` instead of `{}` to avoid unexpected behavior.